### DISCUSSION TITLE
How to clone and make changes in ultralytics

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/ultralytics#11230] [Q&A]
was trying to clone the repo and make some changes to contribute to " specific class object counter ", Although was unable to do so. can anybody explain how to test whether they are working or not. Little about me I am a student who is currently pursuing my Bachelors and a beginner to. this doesn't worked for me

Git clone
Clone the ultralytics repository if you are interested in contributing to the development or wish to experiment with the latest source code. After cloning, navigate into the directory and install the package in editable mode -e using pip.

**Comment by pderrenger** [comment]
Hello! It's great to see your interest in contributing to the Ultralytics project! To clone the repository and test your changes, you can follow these steps:

1. First, clone the repository:
   ```bash
   git clone https://github.com/ultralytics/ultralytics.git
   cd ultralytics
   ```

2. Install the package in editable mode to allow for dynamic updates:
   ```bash
   pip install -e .
   ```

3. After making your changes, you can run the unit tests to ensure everything is functioning correctly:
   ```bash
   pytest
   ```

If you face any issues, it would be helpful to include error messages or specific problems you encounter. Good luck, and I'm here to help if you have more questions! 👍
